src/app/home/tutorial-page/tutorial-page.component.ts

```diff
@@ -1,5 +1,18 @@
 import { Component, OnInit } from '@angular/core';
 import { DomSanitizer, SafeResourceUrl } from '@angular/platform-browser';
+import { Router } from '@angular/router';
+import {
+  ISandbox,
+  SandboxService
+} from 'src/app/services/http-services/sandbox.service';
+import { IProblem } from '../problem-list/problem.model';
+import { DOCUMENT, Location } from '@angular/common';
+import { Inject } from '@angular/core';
+
```

Comment:
Import `IRetryResponse` from `problem.model.ts`. Let's use model file to keep our interfaces.